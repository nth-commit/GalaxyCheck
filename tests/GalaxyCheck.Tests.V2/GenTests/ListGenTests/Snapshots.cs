using GalaxyCheck;
using GalaxyCheck.Gens;
using Snapshooter;
using Snapshooter.Xunit;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using Xunit;

namespace Tests.V2.GenTests.ListGenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IListGen<int>>
            {
                { "Between_Lengths_1_And_4_And_Elements_0_And_10", Gen.Int32().Between(0, 10).ListOf().BetweenLengths(1, 4) },
                { "Between_Lengths_1_And_8_And_Elements_0_And_100", Gen.Int32().Between(0, 100).ListOf().BetweenLengths(1, 8) },
                { "Between_Lengths_1_And_16_And_Elements_0_And_10000", Gen.Int32().Between(0, 10000).ListOf().BetweenLengths(1, 16) },
                { "Between_Defaults", Gen.Int32().ListOf() },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var gen0 = gen.WithLengthBias(bias);

                var sample = gen.Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IListGen<int>>
            {
                { "Between_Lengths_1_And_4_And_Elements_0_And_10", Gen.Int32().Between(0, 10).ListOf().BetweenLengths(1, 4) },
                { "Between_Lengths_1_And_8_And_Elements_0_And_100", Gen.Int32().Between(0, 100).ListOf().BetweenLengths(1, 8) },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var gen0 = gen.WithLengthBias(bias);

                var sample = gen.Advanced
                    .SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => JsonSerializer.Serialize(x));

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
