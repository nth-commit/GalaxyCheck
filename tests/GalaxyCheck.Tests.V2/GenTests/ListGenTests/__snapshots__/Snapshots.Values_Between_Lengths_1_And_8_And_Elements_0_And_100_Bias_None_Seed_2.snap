[
  [
    0
  ],
  [
    1
  ],
  [
    1
  ],
  [
    1
  ],
  [
    0
  ],
  [
    1
  ],
  [
    1
  ],
  [
    0
  ],
  [
    1
  ],
  [
    1
  ],
  [
    1
  ],
  [
    2
  ],
  [
    1
  ],
  [
    2
  ],
  [
    1
  ],
  [
    2
  ],
  [
    0
  ],
  [
    1
  ],
  [
    1
  ],
  [
    2
  ],
  [
    2
  ],
  [
    2
  ],
  [
    1
  ],
  [
    0
  ],
  [
    1
  ],
  [
    0
  ],
  [
    2
  ],
  [
    1
  ],
  [
    3
  ],
  [
    4
  ],
  [
    2
  ],
  [
    1
  ],
  [
    1
  ],
  [
    4
  ],
  [
    1
  ],
  [
    1
  ],
  [
    2
  ],
  [
    1
  ],
  [
    4
  ],
  [
    3
  ],
  [
    1
  ],
  [
    3
  ],
  [
    7
  ],
  [
    6,
    2
  ],
  [
    0
  ],
  [
    8
  ],
  [
    2,
    0
  ],
  [
    1
  ],
  [
    4,
    7
  ],
  [
    8
  ],
  [
    10,
    10
  ],
  [
    1,
    4
  ],
  [
    4
  ],
  [
    11,
    6
  ],
  [
    4
  ],
  [
    0
  ],
  [
    6
  ],
  [
    4
  ],
  [
    2,
    1,
    4,
    0,
    14,
    9,
    3,
    100
  ],
  [
    1,
    6
  ],
  [
    14
  ],
  [
    10
  ],
  [
    11,
    13
  ],
  [
    14
  ],
  [
    18
  ],
  [
    100,
    6
  ],
  [
    14
  ],
  [
    14,
    14,
    1
  ],
  [
    11,
    2,
    11
  ],
  [
    8,
    11,
    21
  ],
  [
    24,
    100,
    0,
    100,
    19,
    13,
    2,
    15
  ],
  [
    11
  ],
  [
    17,
    100
  ],
  [
    9,
    11
  ],
  [
    10,
    15,
    17
  ],
  [
    10,
    100
  ],
  [
    2,
    19,
    21,
    20,
    5,
    100,
    17,
    33
  ],
  [
    25,
    17,
    28
  ],
  [
    8,
    10
  ],
  [
    6,
    18
  ],
  [
    13,
    17,
    20
  ],
  [
    100,
    28,
    100,
    100,
    23,
    23,
    100,
    0
  ],
  [
    27,
    100,
    34,
    100,
    100,
    100,
    100,
    43
  ],
  [
    41,
    7,
    35,
    27,
    42,
    100,
    20,
    45
  ],
  [
    48,
    15,
    100,
    29,
    48,
    100,
    36,
    19
  ],
  [
    21,
    22,
    100,
    25,
    11,
    31,
    29,
    41
  ],
  [
    9,
    11,
    18
  ],
  [
    3
  ],
  [
    10,
    57,
    100
  ],
  [
    100,
    13
  ],
  [
    27,
    35,
    100,
    100,
    100
  ],
  [
    27,
    48,
    100,
    100
  ],
  [
    100,
    25,
    64,
    100
  ],
  [
    37,
    100,
    61,
    100,
    100,
    30,
    33,
    29
  ],
  [
    52
  ],
  [
    57,
    64,
    100,
    33,
    54,
    100
  ],
  [
    100,
    100,
    33,
    100,
    100,
    72,
    45,
    100
  ],
  [
    26,
    100,
    100,
    25,
    100,
    100,
    3,
    86
  ],
  [
    100,
    100,
    59,
    30,
    47,
    100,
    100,
    49
  ],
  [
    28,
    81,
    51,
    100,
    0
  ]
]
